packages/core/src/credential/Credential.ts

```diff
@@ -249,7 +254,9 @@ export class Credential implements ICredential {
     // remove these attributes
     credential.request.removeClaimProperties(excludedClaimProperties)
 
-    const keys = claimerDid.getKeys(KeyRelationship.authentication)
+    const keys = claimerDid.getKeys(
```

Comment:
A custom type guard could be added here. 

```
function isDidVerificationKey(
  key: DidVerificationKey[] | DidKey[]
): key is DidVerificationKey[] {
  if ('BaseDidKey' in key && 'type' in key) {
    return true
  }
  return false
}
```

Reference here https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Add a check   
```
if (!isDidVerificationKey(keys)) throw new Error('false')
```